packages/core/src/app/ovm/deciders/examples/utils.ts

```diff
@@ -0,0 +1,27 @@
+import { ParsedMessage } from '../../../../types/serialization'
+import { objectsEqual } from '../../../utils'
+
+export class Utils {
+  /**
+   * Determines whether or not the provided ParsedMessages conflict.
+   * Conflicting messages have the same channelID and nonce but different data.
+   *
+   * @param message The first message
+   * @param other The second message
+   * @returns True if they conflict, false otherwise
+   */
+  public static messagesConflict(
+    message: ParsedMessage,
+    other: ParsedMessage
+  ): boolean {
+    return (
+      message.message.channelId.equals(other.message.channelId) &&
+      message.message.nonce.equals(other.message.nonce) &&
+      (message.sender.equals(other.sender) ||
+        message.sender.equals(other.recipient)) &&
+      (message.recipient.equals(other.recipient) ||
+        message.recipient.equals(other.sender)) &&
+      !objectsEqual(message.message.data, other.message.data)
+    )
+  }
```

Comment:
Maybe this function should be `stateChannelMessagesConflict(...)`

Reply:
It's probably not going to be used for other messages, so you're right.

In principle, this should work for any type that follows the base `Message` type. The contents of `data` is what determines the type of message (like `StateChannelMessage`), and we don't make any assumptions on that in this function.

Reply:
Updated for good measure 😄